OBSDOCS#573: add docs for configuring resource limits and requests for monitoring components (#69270)
openshift/openshift-docs: modules/monitoring-specifying-limits-and-requests-for-monitoring-components.adoc

// Module included in the following assemblies:
//
// * monitoring/configuring-the-monitoring-stack.adoc

:_mod-docs-content-type: PROCEDURE
[id="specifying-limits-and-resource-requests-for-monitoring-components_{context}"]
= Specifying limits and requests for monitoring components 

> To configure CPU and memory resources, specify values for resource limits and requests for each monitoring component in the `cluster-monitoring-config` config map object in the default `openshift-monitoring` namespace.

Reviewer: ```suggestion
To configure CPU and memory resources, specify values for resource limits and requests for each monitoring component in the `cluster-monitoring-config` config map object in the `openshift-monitoring` namespace.
```

Author: removed

          memory: 500Mi
    prometheusOperator:
      resources:
        limits:
          cpu: 500m
          memory: 1Gi
        requests:
          cpu: 200m
          memory: 500Mi
    k8sPrometheusAdapter:
      resources:
        limits:
          cpu: 500m
          memory: 1Gi
        requests:
          cpu: 200m
          memory: 500Mi
    kubeStateMetrics:
      resources:
        limits:
          cpu: 500m
          memory: 1Gi
        requests:
          cpu: 200m
          memory: 500Mi
    telemeterClient:
      resources:
        limits:
          cpu: 500m
          memory: 1Gi
        requests:
          cpu: 200m
          memory: 500Mi
    openshiftStateMetrics:
      resources:
        limits:
          cpu: 500m
          memory: 1Gi
        requests:
          cpu: 200m
          memory: 500Mi
    thanosQuerier:
      resources:
        limits:
          cpu: 500m
          memory: 1Gi
        requests:
          cpu: 200m
          memory: 500Mi
    nodeExporter:
      resources:
        limits:
          cpu: 50m
          memory: 150Mi
        requests:
          cpu: 20m
          memory: 50Mi
    monitoringPlugin:
      nodeSelector:

>         node-role.kubernetes.io/control-plane: ""

Reviewer: ```
      nodeSelector:
        node-role.kubernetes.io/control-plane: ""
```
is no need

Author: removed

      resources:
        limits:
          cpu: 500m
          memory: 1Gi
        requests:
          cpu: 200m
          memory: 500Mi
    prometheusOperatorAdmissionWebhook:
      resources:
        limits:
          cpu: 50m
          memory: 100Mi
        requests:
          cpu: 20m
          memory: 50Mi

> ----

Reviewer: I am kind think we don't need to give example for every component, list a few is fine

Author: I will make clear that the config procedure is for core platform monitoring and the UWM config can also be done, if desired, but I think it'd be redundant to do an almost identical procedure only for UWM.

// Module included in the following assemblies:
//
// * monitoring/configuring-the-monitoring-stack.adoc

:_mod-docs-content-type: PROCEDURE
[id="specifying-limits-and-resource-requests-for-monitoring-components_{context}"]
= Specifying limits and requests for monitoring components 

To configure CPU and memory resources, specify values for resource limits and requests in the appropriate `ConfigMap` object for the namespace in which the monitoring component is located:

> * The `cluster-monitoring-config` config map in the `openshift-monitoring` namespace for core platform monitoring.

Reviewer 2: ```suggestion
* The `cluster-monitoring-config` config map in the `openshift-monitoring` namespace for core platform monitoring
```
I don't think this fragment needs ending punctuation

Author: Removed

> * The `user-workload-monitoring-config` config map in the `openshift-user-workload-monitoring` namespace for components that monitor user-defined projects.

Reviewer 2: ```suggestion
* The `user-workload-monitoring-config` config map in the `openshift-user-workload-monitoring` namespace for components that monitor user-defined projects
```
Same here -- you can omit the period

Author: Removed

.Prerequisites

* *If you are configuring core platform monitoring components*:
** You have access to the cluster as a user with the `cluster-admin` cluster role.

> ** You have created the `cluster-monitoring-config` `ConfigMap` object.

Reviewer 2: ```suggestion
** You created a `ConfigMap` object named `cluster-monitoring-config`.
```
I tried to separate `ConfigMap` from `cluster-monitoring-config` so that the two monospace terms wouldn't be misinterpreted.

Author: Excellent suggestion. Changed.

* *If you are configuring components that monitor user-defined projects*:
** You have access to the cluster as a user with the `cluster-admin` cluster role, or as a user with the `user-workload-monitoring-config-edit` role in the `openshift-user-workload-monitoring` project.

> * You have installed the OpenShift CLI (`oc`).

Reviewer 2: ```suggestion
* You installed the OpenShift CLI (`oc`).
```

Author: `You have installed` is what's used throughout the monitoring documentation, so I'm leaving this as-is to remain consistent.

.Procedure

> . To configure core platform monitoring components, edit the `cluster-monitoring-config` config map object in the `openshift-monitoring` namespace:

Reviewer 2: ```suggestion
. To configure core platform monitoring components, edit the `cluster-monitoring-config` object in the `openshift-monitoring` namespace:
```
Because you've already mentioned that `cluster-monitoring-config` is a `ConfigMap` object, would it be safe to remove "config map" here?

Author: good thought, but I'm going to leave it there in case a user only skimmed the earlier content. It's important that they know it's a config map object.

+
[source,terminal]
----
$ oc -n openshift-monitoring edit configmap cluster-monitoring-config
----

. Add values to define resource limits and requests for each core platform monitoring component you want to configure.
+

> [WARNING]

Reviewer 2: Per the SSG, a warning is appropriate for "Information about potential system damage, data loss, or a support-related issue if the user disregards this admonition. Explain the problem, cause, and offer a solution that works. If available, offer information to avoid the problem in the future or state where to find more information."

If the value set for a limit is not higher than the value set for a request, will that cause system damage, data loss, or a support-related issue? If not, consider changing the warning to an IMPORTANT admonition.

Author: Agree. Changing to an Important admonition. Good catch! Thanks.

    kubeStateMetrics:
      resources:
        limits:
          cpu: 500m
          memory: 1Gi
        requests:
          cpu: 200m
          memory: 500Mi
    telemeterClient:
      resources:
        limits:
          cpu: 500m
          memory: 1Gi
        requests:
          cpu: 200m
          memory: 500Mi
    openshiftStateMetrics:
      resources:
        limits:
          cpu: 500m
          memory: 1Gi
        requests:
          cpu: 200m
          memory: 500Mi
    thanosQuerier:
      resources:
        limits:
          cpu: 500m
          memory: 1Gi
        requests:
          cpu: 200m
          memory: 500Mi
    nodeExporter:
      resources:
        limits:
          cpu: 50m
          memory: 150Mi
        requests:
          cpu: 20m
          memory: 50Mi
    monitoringPlugin:
      resources:
        limits:
          cpu: 500m
          memory: 1Gi
        requests:
          cpu: 200m
          memory: 500Mi
    prometheusOperatorAdmissionWebhook:
      resources:
        limits:
          cpu: 50m
          memory: 100Mi
        requests:
          cpu: 20m
          memory: 50Mi
----

. Save the file to apply the changes automatically.
+

> [WARNING]

Reviewer 2: See my earlier comment about whether to use a WARNING or IMPORTANT admonition.

Author: Agree. Changing to an Important admonition.

:_mod-docs-content-type: PROCEDURE
[id="specifying-limits-and-resource-requests-for-monitoring-components_{context}"]
= Specifying limits and requests for monitoring components 

To configure CPU and memory resources, specify values for resource limits and requests in the appropriate `ConfigMap` object for the namespace in which the monitoring component is located:

* The `cluster-monitoring-config` config map in the `openshift-monitoring` namespace for core platform monitoring.
* The `user-workload-monitoring-config` config map in the `openshift-user-workload-monitoring` namespace for components that monitor user-defined projects.

.Prerequisites

* *If you are configuring core platform monitoring components*:
** You have access to the cluster as a user with the `cluster-admin` cluster role.
** You have created the `cluster-monitoring-config` `ConfigMap` object.
* *If you are configuring components that monitor user-defined projects*:
** You have access to the cluster as a user with the `cluster-admin` cluster role, or as a user with the `user-workload-monitoring-config-edit` role in the `openshift-user-workload-monitoring` project.

> ** You have created the `user-workload-monitoring-config` `ConfigMap` object.

Reviewer: since 4.11, `user-workload-monitoring-config` configmap is created by default if enabled UWM

Author: Because the config map exists by default, I think we can just cut it entirely.
